v2/internal/frontend/desktop/linux/window.go

```diff
@@ -549,6 +549,20 @@ void DisableContextMenu(void* webview) {
 	g_signal_connect(WEBKIT_WEB_VIEW(webview), "context-menu", G_CALLBACK(disableContextMenu), NULL);
 }
 
+void SetWindowIcon(GtkWindow* window, const guchar* buf, gsize len) {
+	GdkPixbufLoader* loader = gdk_pixbuf_loader_new();
+	if (!loader) {
+		return;
+	}
+	if (!gdk_pixbuf_loader_write(loader, buf, len, NULL) && !gdk_pixbuf_loader_close(loader, NULL)) {
+		return;
+	}
+	GdkPixbuf* pixbuf = gdk_pixbuf_loader_get_pixbuf(loader);
+	if (!pixbuf) {
+		return;
+	}
+	gtk_window_set_icon(window, pixbuf);
+}
```

Comment:
freeing the loader cause a 'double free or corruption' error and a SIGABRT, still for me it seems logical to free it after usage... anyway even if it leaks some memory, onetime...

Reply:
Yeah that's weird :-)

Reply:
do you see a problem with that?

Reply:
thanks for the hint